src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs

```diff
@@ -512,11 +512,13 @@ private int ReadingNotSupported(byte[] destination, int offset, int count)
 		/// <returns>The actual number of bytes read.</returns>
 		private int InitialRead(byte[] destination, int offset, int count)
 		{
-			if (!CanDecompressEntry)
+			if (entry == null)
```

Comment:
I don't think it should be possible to get here in that case? (maybe  auseful sanity check in case though)

Reply:
No, I don't think so either, but since the `CanDecompressEntry` used to check it, I didn't want to remove it since the rest of the body dereferences entity a lot. It shouldn't be too hard to verify though.

Reply:
I was going to query if changing this effects the handling of AES entries (where the 'entry.IsCrypted' path only handles ZipCrypto entries), but I don't *think* it can get this far for those because of the earlier checks in GetNextEntry.

Reply:
Yeah, exactly. There is only one reference to the method, and that method sets and dereferences entity just before invoking this one. Hence I removed the check.